Mathlib/RingTheory/Spectrum/Prime/Homeomorph.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2025 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.FieldTheory.PurelyInseparable.Basic
+import Mathlib.RingTheory.Flat.Basic
+import Mathlib.RingTheory.Spectrum.Prime.Topology
+
+/-!
+# Purely inseparable extensions are universal homeomorphisms
+
+If `K` is a purely inseparable extension of `k`, the induced map `Spec K ⟶ Spec k` is a universal
+homeomorphism, i.e. it stays a homeomorphism after arbitrary base change.
+
+## Main results
+
+- `PrimeSpectrum.isHomeomorph_comap`: if `f : R →+* S` is a ring map with locally nilpotent kernel
+  such that for every `x : S`, there exists `n > 0` such that `x ^ n` is in the image of `f`,
+  `Spec f` is a homeomorphism.
+- `PrimeSpectrum.isHomeomorph_comap_of_isPurelyInseparable`: `Spec K ⟶ Spec k` is a universal
+  homeomorphism for a purely inseparable field extension `K` over `k`.
+-/
+
+open TensorProduct
+
+variable (k K R S : Type*) [Field k] [Field K] [Algebra k K] [CommRing R] [Algebra k R] [CommRing S]
+
+variable {R S} in
+@[stacks 0BR8 "Homeomorphism part"]
+lemma PrimeSpectrum.isHomeomorph_comap (f : R →+* S) (H : ∀ (x : S), ∃ n > 0, x ^ n ∈ f.range)
+    (hker : RingHom.ker f ≤ nilradical R) : IsHomeomorph (PrimeSpectrum.comap f) := by
```

Comment:
This is golfed [here](https://github.com/leanprover-community/mathlib4/commit/111b0b94134edb47bb262c37cca8b906b6e08078).